rita_common/src/usage_tracker/mod.rs

```diff
@@ -133,44 +162,94 @@ impl UsageTracker {
             payments: VecDeque::new(),
         };
 
-        match file {
-            Ok(mut file) => {
-                let mut byte_contents = Vec::new();
-                // try compressed
-                match file.read_to_end(&mut byte_contents) {
-                    Ok(_bytes_read) => {
-                        let decoder = ZlibDecoder::new(&byte_contents[..]);
-                        // Extract data from decoder, note this is streaming decoding, so we're
-                        // decompressing the data as we take it out of the zlib compressed object
-                        trace!("attempting to unzip or read bw history");
-                        let deserialized: Result<UsageTracker, SerdeError> =
-                            serde_json::from_reader(decoder);
-
-                        match deserialized {
-                            Ok(value) => value,
-                            Err(e) => {
-                                error!(
-                                    "Failed to deserialize bytes in compressed bw history {:?}",
-                                    e
-                                );
-                                blank_usage_tracker
-                            }
-                        }
-                    }
+        let file_exists =
+            Path::new(&settings::get_rita_common().network.usage_tracker_file).exists();
+        let file_is_bincode = Path::new(&settings::get_rita_common().network.usage_tracker_file)
+            .extension()
+            .unwrap()
+            .eq("bincode");
+        fn decompressed(mut file: File) -> Result<Vec<u8>, RitaCommonError> {
+            let mut byte_contents = Vec::new();
+            match file.read_to_end(&mut byte_contents) {
+                Ok(_bytes_read) => {
+                    let mut decoder = ZlibDecoder::new(&byte_contents[..]);
+                    let mut bytes = Vec::<u8>::new();
+                    decoder.read_to_end(&mut bytes)?;
+
+                    Ok(bytes)
+                }
+                Err(e) => {
+                    error!("Failed to read usage tracker file! {:?}", e);
+                    Err(RitaCommonError::StdError(e))
+                }
+            }
+        }
+        let file = File::open(settings::get_rita_common().network.usage_tracker_file);
+        let fileopen = match file {
+            Ok(file) => file,
+            Err(e) => {
+                error!("Failed to open usage tracker file! {:?}", e);
+                return blank_usage_tracker;
+            }
+        };
+        match (file_exists, file_is_bincode, decompressed(fileopen)) {
```

Comment:
nice! I like the match, please comment each of these cases with a quick plain English summary 

// New file exists, no need for conversion
// New file does not exist, old file does, we must load using the old format and update settings 

etc etc